tests/import_specifications/test_file_parser.py

```diff
@@ -42,39 +115,69 @@ def test_Error_init_w_OTHER_success():
 
     assert e.error == ErrorType.OTHER
     assert e.message == "foo"
-    assert e.source == None
+    assert e.source_1 is None
+    assert e.source_2 is None
 
     # all
-    e = Error(ErrorType.OTHER, message="foo", source=SpecificationSource("whoo"))
+    e = Error(ErrorType.OTHER, message="foo", source_1=spcsrc("wooo"))
 
     assert e.error == ErrorType.OTHER
     assert e.message == "foo"
-    assert e.source == SpecificationSource("whoo")
+    assert e.source_1 == spcsrc("wooo")
+    assert e.source_2 is None
 
 
 def test_Error_init_fail():
-    error_init_fail(ErrorType.FILE_NOT_FOUND, None, None, ValueError(
-        "source is required for a FILE_NOT_FOUND error"))
-    err = "source and message are required for a PARSE_FAIL error"
-    error_init_fail(ErrorType.PARSE_FAIL, None, SpecificationSource("foo"), ValueError(err))
-    error_init_fail(ErrorType.PARSE_FAIL, "msg", None, ValueError(err))
-    error_init_fail(ErrorType.OTHER, None, None, ValueError(
+    error_init_fail(None, None, None, None, ValueError("error is required"))
```

Comment:
It might be easier for the reader to know what's going on if they haven't memorized the required inputs for error_init_fail by adding in keyword args

Reply:
The reason I don't use kw args in this type of test is because they'd be repeated over and over for every line and take up a lot of space with redundant information. My general feeling is that the error checking function is right below this one and so isn't too hard to reference, but there are a lot of test cases packed into this function so I also added a comment to the beginning of the test listing the arguments in order. Is that good enough?

Reply:
Instead of adding it into a comment, you could add it into the first error_init_fail call and then you wouldn't have to worry about updating the comment if the docs change. 

My personal preference is to use kw args for any calls taking more than 1 or 2 inputs, and it's easier to see what's going on for kw args personally. We don't have any established code standard here nor am I requesting changes in this PR, looks good enough for me.

Reply:
Mmm... my feeling is that the function being called is right below these calls, and with typehints down there and the args in the comment it's good enough. I've been doing it this way for 5 years w/o problems (and also in Java, where you can't give keyword args)